scvi/models/modules.py

```diff
@@ -94,15 +184,43 @@ def forward(self, dispersion, z, library, *cat_list):
 
 # Decoder
 class Decoder(nn.Module):
-    def __init__(self, n_input, n_output, n_cat_list=[], n_layers=1, n_hidden=128, dropout_rate=0.1):
+    r"""Decodes data from latent space of ``n_input`` dimensions to ``n_output``
+    dimensions using a fully-connected neural network of ``n_hidden`` layers.
+    Output is the mean and variance of a multivariate Gaussian
+
+    :param n_input: The dimensionality of the input (latent space)
+    :param n_output: The dimensionality of the output (data space)
+    :param n_cat_list: A list containing the number of categories
+                       for each category of interest. Each category will be
+                       included using a one-hot encoding
+    :param n_layers: The number of fully-connected hidden layers
+    :param n_hidden: The number of nodes per hidden layer
+    :param dropout_rate: Dropout rate to apply to each of the hidden layers
+    """
+
+    def __init__(self, n_input: int, n_output: int,
+                 n_cat_list: Iterable[int] = None, n_layers: int = 1,
+                 n_hidden: int = 128, dropout_rate: float = 0.1):
         super(Decoder, self).__init__()
-        self.decoder = FCLayers(n_in=n_input, n_out=n_hidden, n_cat_list=n_cat_list, n_layers=n_layers,
+        self.decoder = FCLayers(n_in=n_input, n_out=n_hidden,
+                                n_cat_list=n_cat_list, n_layers=n_layers,
                                 n_hidden=n_hidden, dropout_rate=dropout_rate)
 
         self.mean_decoder = nn.Linear(n_hidden, n_output)
         self.var_decoder = nn.Linear(n_hidden, n_output)
 
-    def forward(self, x, *cat_list):
+    def forward(self, x: torch.Tensor, *cat_list: int):
+        r"""The forward computation for a single sample.
+
+         #. Decodes the data from the latent space using the decoder network
+         #. Returns mean and variance for a multivariate Gaussian
```

Comment:
I think it should be parameters for a ZINB (zero inflated negative binomial) output distribution

Reply:
Well, it can be used several ways -- maybe better just to say it outputs parameters for a distribution.

Reply:
Yeah this was one place I was looking for clarification. This model is used in [SVAEC](https://github.com/YosefLab/scVI/blob/master/scvi/models/svaec.py#L56) as `decoder_z1_z2` which I interpreted as translating from one latent space to another (I think), and the results are passed into a normal distribution but as you say it could have other uses.